text/0097-unbonding_queue.md

```diff
@@ -0,0 +1,165 @@
+
+This extension and further specifications are left out of this RFC, because it adds further complexity and the empirical analysis above suggests that average unbonding times will already be close the `LOWER_BOUND`, making a more complex design unnecessary. We advise to first implement the discussed mechanism and assess after some experience whether an extension is desirable.
+
+## Drawbacks
+
+- **Lower security for LRAs:** Without a doubt, the theoretical security against LRAs decreases. But, as we argue, the attack is still costly enough to deter attacks and the attack is sufficiently theoretical. Here, the benefits outweigh the costs.
+- **Griefing attacks:** A large holder could pretend to unbond a large amount of their tokens to prevent other users to exit the network earlier. This would, however be costly due to the fact that the holder loses out on staking rewards. The larger the impact on the queue, the higher the costs. In any case it must be noted that the `UPPER_BOUND` is still 28 days, which means that nominators are never left with a longer unbonding period than currently. There is not enough gain for the attacker to endure this cost.
+- **Challenge for Custodians and Liquid Staking Providers**: Changing the unbonding time, especially making it flexible, requires entities that offer staking derivatives to rethink and rework their products.
+
+## Testing, Security, and Privacy
+
+NA
+
+## Performance, Ergonomics, and Compatibility
+
+NA
+
+### Performance
+
+The authors cannot see any potential impact on performance.
+
+### Ergonomics
+
+The authors cannot see any potential impact on ergonomics for developers. We discussed potential impact on UX/UI for users above.
+
+### Compatibility
+
+The authors cannot see any potential impact on compatibility. This should be assessed by the technical fellows.
+
+
+### Prior Art and References
+- Ethereum proposed a [similar solution](https://blog.stake.fish/ethereum-staking-all-you-need-to-know-about-the-validator-queue/)
+- Alistair did some initial [write-up](https://hackmd.io/SpzFSNeXQM6YScW1iODC_A)
+- There are [other solutions](https://arxiv.org/pdf/2208.05408.pdf) that further mitigate the risk of LRAs.
+
+### The Unresolved Question: Deferred slashing
+
+Currently we defer applying many slashes until 28 days have passed. This was implemented so we can conveniently cancel slashes via governance in the case that the slashing was due to a bug. While rare on Polkadot, such bugs cause a significant fraction of slashes. This includes slashing for attacks other than LRAs for which we've assumed that 2 days is enough to slash. But 2 days in not enough to cancel slashes via OpenGov.
```

Comment:
https://github.com/polkadot-fellows/runtimes/blob/7c69345c75910006d56e60e2f9a93c9d0f44f280/relay/polkadot/src/lib.rs#L654
```suggestion
Currently we defer applying many slashes until 27 days have passed. This was implemented so we can conveniently cancel slashes via governance in the case that the slashing was due to a bug. While rare on Polkadot, such bugs cause a significant fraction of slashes. This includes slashing for attacks other than LRAs for which we've assumed that 2 days is enough to slash. But 2 days in not enough to cancel slashes via OpenGov.
```

Reply:
Right now there is a slash that happened in era 1498 which, polkadot.js apps says, will be applied in era 1526 (28 eras later). Not sure where the discrapency comes from, but since we are talking about days here anyway, I'll just say that it takes *around 28 days*, which should be fine enough.